docs/frontend.md

```diff
@@ -1,5 +1,44 @@
 # Front-end Architecture
 
+While the Login.gov application is architected as a Ruby on Rails project and most of its pages are
+fully rendered by the backend server, the front end is set up to use modern best-practices to suit
+the needs of interactivity and consistency in the user experience, and to ensure a maintainable and
+convenient developer experience.
+
+As a high-level overview, the front end consists of:
+
+- [Propshaft](https://github.com/rails/propshaft) is used as an asset pipeline library for Ruby on
+  Rails, in combination with [cssbundling-rails](https://github.com/rails/cssbundling-rails) and
+  [jsbundling-rails](https://github.com/rails/jsbundling-rails) for compiling stylesheets and
+  scripts.
+- JavaScript is written as [TypeScript](https://www.typescriptlang.org/), bundled using [Webpack](https://webpack.js.org/)
+  with syntax transformation applied by [Babel](https://babeljs.io/).
+   - For highly-interactive pages, we use [React](https://react.dev/)
+   - For all other JavaScript interaction, we use native [web components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components)
+     (custom elements)
+- Stylesheets are written as [Sass](https://sass-lang.com/), and builds upon the [Login.gov Design System](https://github.com/18F/identity-design-system),
+  which in turn builds upon the [U.S. Web Design System (USWDS)](https://designsystem.digital.gov/).
+- HTML reuse is facilitated by the [ViewComponent](https://viewcomponent.org/) gem, enhanced with
+  support for per-component JavaScript and stylesheet "sidecar" assets.
+
+The general folder structure for front-end assets includes:
+
+```
+app/
+├─ assets/
+│  ├─ builds/      # Source location for compiled stylesheets used by Propshaft in asset compilation
+│  ├─ fonts/       # Source font assets
+│  ├─ images/      # Source image assets
+│  └─ stylesheets/ # Source Sass files
+├─ components/     # ViewComponent implementations
+└─ javascript/
+   ├─ packages/    # JavaScript workspace NPM packages
+   └─ packs/       # JavaScript entrypoints referenced by pages
```

Comment:
maybe we should link the Packs README since that has more info on how to use them and what files belong there?

Reply:
We do link to it, though much further down in the documentation (under ["Additional Resources"](https://github.com/18F/identity-idp/blob/main/docs/frontend.md#additional-resources)). Though I agree, I was trying to think about how to present this in a way that's easy to understand at a glance. I don't know that I can easily make the text itself linkable, though we could add a link after the text (maybe what you had in mind). Another option could be a nested bulleted list, similar to how [`rails-directory-structure-guide`](https://github.com/jwipeout/rails-directory-structure-guide?tab=readme-ov-file#directory-structure) is laid out.

Reply:
Yeah bulleted list makes sense, so more like this?

- app
    - assets
    - javascript
        - [packs](README.md): JavaScript entrypoints referenced by pages